scripts-available/CDB_GhostTables.sql

```diff
@@ -0,0 +1,104 @@
+-- Table to store the transaction id from DDL events to avoid multiple executions
+CREATE TABLE IF NOT EXISTS cartodb.cdb_ddl_execution(txid integer PRIMARY KEY);
+
+-- Enqueues a job to run Ghost tables linking process for the provided user_id
+CREATE OR REPLACE FUNCTION _CDB_LinkGhostTables(user_id text) 
+RETURNS void
+AS $$
+  if not user_id:
+    return
+
+  client = GD.get('redis', None)
+
+  retry = 3
+  error = ''
+  redis_host = '127.0.0.1'
+  redis_port = 6379
+  redis_timeout = 5
+
+  while True:
+
+    if not client:
+        try:
+          import redis
+          client = GD['redis'] = redis.Redis(host=redis_host, port=redis_port, socket_timeout=redis_timeout)
+        except Exception as err:
+          error = "client_error - %s" % str(err)
+          # NOTE: no retries on connection error
+          plpy.error('Ghost tables connection error: ' +  str(err))
+          break
+
+    try:
+      job = '{{"class":"Resque::UserDBJobs::UserDBMaintenance::LinkGhostTables","args":["{}"]}}'.format(user_id)
+      client.rpush("resque:queue:user_dbs", job)
+      break
+    except Exception as err:
+      error = "request_error - %s" % str(err)
+      client = GD['redis'] = None # force reconnect
+      if not retry:
+        plpy.error('Ghost tables error: ' +  str(err))
+        break
+      retry -= 1 # try reconnecting
+$$ LANGUAGE 'plpythonu' VOLATILE PARALLEL UNSAFE;
+
+-- Enqueues a job to run Ghost tables linking process for the current user
+CREATE OR REPLACE FUNCTION CDB_LinkGhostTables()
+RETURNS void
+AS $$
+  DECLARE
+    user_id TEXT;
+  BEGIN
+    EXECUTE 'SELECT (regexp_match(session_user, ''[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}''))[1];' INTO user_id;
```

Comment:
I've been thinking about this and maybe, to avoid the problems derived from the user of regexp (for example this could not work with regular apis), we should send the username to the TIS and create another Job in cartodb that:
- Get the ID from username
- Call the ghost table task with that ID